### PR TITLE
Check for wildcard before running `plz test -d`

### PR DESCRIPTION
**Pull request by windrunner99**
My attempt of fixing issue #2204

### CODE REVIEW

File: src/please.go [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ func Please(targets []core.BuildLabel, config *core.Configuration, shouldBuild,
 		}
 	}
 
-	if state.DebugFailingTests && len(targets) != 1 {
+	if state.DebugFailingTests && (len(targets) != 1 || (len(targets) == 1 && targets[0].Name == "...")) {
```
Comment (peterebden) [inline src/please.go:1064]: This should be `(targets[0].IsAllSubpackages() || targets[0].IsAllTargets())`; that picks up `:all` as well.

We could also add a method to check for either of those two, there are other places this has come up too.

**Review by peterebden — COMMENTED** [review]
Thanks!

File: src/please.go (hunk 1 above)
Comment (windrunner99) [reply src/please.go]: Used the functions liked you said. The logic now looks a little bit messy so I added a comment.
But you are right, adding a Method that checks if either of those two are true to check if the BuildLable potentially refers more than one rule wild clear this up.
Let me think of a descriptive name and I will add it to this PR.

Comment (peterebden) [reply src/please.go]: `IsPseudoTarget` is the best I can think of to capture both of these cases where it's not actually referring to a real target.

Thanks!

**Comment by peterebden** [comment]
Thanks!

**Pull request merged**